src/Diagnostics.Logger/ApiMetricsLogger.cs

```diff
@@ -0,0 +1,183 @@
+﻿using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Primitives;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Diagnostics.Logger
+{
+    public class ApiMetricsLogger
+    {
+        private string _requestId;
+        private string _address;
+        private string _verb;
+        private DateTime _startTime;
+        private DateTime _endTime;
+        private long _latencyInMs;
+        private string _subscriptionId;
+        private string _resourceGroupName;
+        private string _resourceName;
+
+        public ApiMetricsLogger(HttpContext context)
+        {
+            StartMetricCapture(context);
+        }
+
+        private void StartMetricCapture(HttpContext context)
+        {
+            if (context.Request.Headers.TryGetValue(HeaderConstants.RequestIdHeaderName, out StringValues values))
+            {
+                _requestId = values.ToString();
+            }
+
+            _startTime = DateTime.UtcNow;
+            _verb = context.Request.Method;
+            _address = context.Request.Path.ToString();
+
+            ParseFromAddress(context);
+        }
+
+        private void ParseFromAddress(HttpContext httpContext)
+        {
+            try
+            {
+                if (String.IsNullOrWhiteSpace(httpContext.Request.Path.ToString()))
+                {
+                    return;
+                }
+
+                var parts = httpContext.Request.Path.ToString().ToLowerInvariant().Split('?')[0].Split(new[] { '/' }, StringSplitOptions.RemoveEmptyEntries);
+                var cursor = ((IEnumerable<string>)parts).GetEnumerator();
+                while (cursor.MoveNext())
```

Comment:
Instead of this way of parsing what you want from the request path, consider using regular expressions and match groups. https://www.dotnetperls.com/regex-groups

Example: subscriptions/SomeGuid/resourceGroups/SomeName/providers/Microsoft.Web/sites/SomeSite

You can have a regex like `"subscribtions/(.*)/resourceGroups/(.*)/providers/Microsoft.Web/sites/(.*)"` and each of these can match to a group and you basically just say `_subscriptionId = Regex.Group(1)`